osu.Game.Rulesets.Taiko/Skinning/Legacy/TaikoLegacySkinTransformer.cs

```diff
@@ -152,32 +152,35 @@ private string getHitName(TaikoSkinComponents component)
             throw new ArgumentOutOfRangeException(nameof(component), $"Invalid component type: {component}");
         }
 
-        public override ISample GetSample(ISampleInfo sampleInfo) => Source.GetSample(new LegacyTaikoSampleInfo(sampleInfo));
+        public override ISample GetSample(ISampleInfo sampleInfo)
+        {
+            if (sampleInfo is HitSampleInfo hitSampleInfo)
+                return Source.GetSample(new LegacyTaikoSampleInfo(hitSampleInfo));
+
+            return base.GetSample(sampleInfo);
+        }
 
         public override IBindable<TValue> GetConfig<TLookup, TValue>(TLookup lookup) => Source.GetConfig<TLookup, TValue>(lookup);
 
-        private class LegacyTaikoSampleInfo : ISampleInfo
+        private class LegacyTaikoSampleInfo : HitSampleInfo
```

Comment:
Without this being a `HitSampleInfo` it doesn't get the correct legacy lookups. See https://github.com/ppy/osu/blob/2e2281c7d22345164d314464341138d3d544f250/osu.Game/Skinning/LegacySkin.cs#L510-L511

Without this, the fallback to `hitnormal` does not occur. It only used to work due to some weird luck with component naming.